### PR TITLE
Remove wp-content directory requirement

### LINKED ISSUE #18: Fix issue with hard coded WP Content location.
In `class-plugin-base.php`, `wp-content` is hard coded and make it impossible to include a plugin anywhere outside of a directory not named `wp-content` (which can be the case when using custom `WP_CONTENT_DIR` or `WPMU_PLUGIN_DIR`).

### PR DESCRIPTION
**Pull request by lukecarbis**
Resolves #18

### CODE REVIEW

File: php/class-plugin-base.php [hunk 1]
```diff
@@ -141,20 +141,14 @@ public function locate_plugin() {
 		if ( '/' !== \DIRECTORY_SEPARATOR ) {
 			$file_name = str_replace( \DIRECTORY_SEPARATOR, '/', $file_name ); // Windows compat.
 		}
-		$plugin_dir = preg_replace( '#(.*plugins[^/]*/[^/]+)(/.*)?#', '$1', $file_name, 1, $count );
-		if ( 0 === $count ) {
```
Comment (kasparsd) [inline php/class-plugin-base.php]: @lukecarbis Did you mean to remove this too?

**Comment by kopepasah** [comment]
@lukecarbis I cannot think of a case where the content directory (`WP_CONTENT_DIR`) constant and the content URL (`WP_CONTENT_URL`) constant would be in a different location, so this seems okay to me. However, I am wondering why the use of `content_url()` over `WP_CONTENT_DIR` in this context.

**Comment by lukecarbis** [comment]
@kopepasah The reason I didn't use `WP_CONTENT_DIR` is because I feel it would be best practice to include an `if( defined( WP_CONTENT_DIR ) )` check first. And if it isn't defined, I'd likely fall back to `content_url()` anyway. Also, there's already precedent for using `content_url()` in this method.

File: php/class-plugin-base.php (hunk 1 above)
Comment (lukecarbis) [inline php/class-plugin-base.php]: Yes. I don't see the point of forcing the plugin to be inside a `plugins` or `mu-plugins` directory. Although I can't imagine a situation where having a plugin elsewhere would actually be the case, enforcing this is simply not required.

I've removed the `preg_replace` in c48cd3af9849159155155dae235c17e8474db557 in favour of double `dirname`. This makes the assumption that the `plugin-base.php` file will remain in a child directory of the main plugin. I think that it's okay to make this assumption for now, especially if we will be pursuing the ideas proposed in #12 later on.

**Comment by westonruter** [comment]
Note that `WP_CONTENT_DIR` will always be defined. See https://github.com/WordPress/wordpress-develop/blob/0b5b1a667f28c76cbc9db3d2a16ced99353f6f77/src/wp-includes/default-constants.php#L64-L65

**Comment by lukecarbis** [comment]
Happy to change it if there's a good reason to. Otherwise, any reason why we shouldn't merge?

**Pull request merged**